app/models/conversation.rb

```diff
@@ -0,0 +1,43 @@
+# == Schema Information
+#
+# Table name: conversations
+#
+#  id         :bigint           not null, primary key
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Conversation < ApplicationRecord
+  has_many :conversation_users, dependent: :destroy
+  has_many :users, through: :conversation_users
+  has_many :messages, dependent: :destroy
+
+  def self.create_conversations(matching_users)
+    user = matching_users.pop
```

Comment:
So, matching_users include the user and all the matches for the user? If yes, don't you think is better to separate them in the params, so it's clearer? `self.create_conversations(user, matching_users)`